src/index.js

```diff
@@ -57,14 +57,30 @@ export default class ScrollBehavior {
     });
   }
 
+  getSavedScrollTarget(key) {
+    const location = this._getCurrentLocation();
+    if (location.action === 'PUSH') {
+      return null;
+    }
+    return this._stateStorage.read(location, key);
+  }
+
   registerElement(key, element, shouldUpdateScroll, context) {
     invariant(
       !this._scrollElements[key],
       'ScrollBehavior: There is already an element registered for `%s`.',
       key,
     );
 
-    this._scrollElements[key] = { element, shouldUpdateScroll };
+    const onElementScroll = () => {
+      this._saveElementPosition(key);
```

Comment:
I think we should rAF-throttle these. They're still writing to local storage. Note in that case you'd have to apply the same logic then to cancel pending timers to save element positions.

Reply:
see updated code